aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -1383,6 +1386,20 @@ internal class RealAWSCognitoAuthPlugin(
         )
     }
 
+    private fun dispatchHubEvent(authState: AuthState) {
+        when {
+            authState.authNState is AuthenticationState.SignedIn -> AuthChannelEventName.SIGNED_IN
+            authState.authNState is AuthenticationState.SignedOut -> AuthChannelEventName.SIGNED_OUT
+            authState.authZState?.deleteUserState is DeleteUserState.UserDeleted -> AuthChannelEventName.USER_DELETED
+            else -> null
+        }?.takeUnless {
+            it != lastPublishedHubEventName.get()
```

Comment:
If current authState == new authState then [listeners](https://github.com/aws-amplify/amplify-android/blob/dev-preview/aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt#L188) are not invoked. I don't think you need to keep track of previous event. Maybe test it out once to confirm.

Reply:
Yes, I added this case after observing that. Technically, Auth states will be different when either of `authN` or `authZ` changes right? So when say, `authZ` changes while `authN` stays as `SignedIn`, listeners will still be invoked.